centralizedmetrics/sqlite_persistence.go

```
@@ -98,6 +98,26 @@ func (r *SQLiteMetricRepository) Delete(metrics []common.Metric) error {
 	return tx.Commit()
 }
 
+func (r *SQLiteMetricRepository) CleanupOldMetrics() error {
+	tx, err := r.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
```

This error is never returned, so either update the line below to `return err` or just name the result parameter `func (r *SQLiteMetricRepository) CleanupOldMetrics() (err error) {`.

---

This is how all the code in persistence files in status-go are. So they are either all broken or there is some magic happening that I don't know about :smile: